modules/@ergo/activity-logs/index.js

```diff
@@ -2,59 +2,28 @@
  * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
  * See LICENSE for license details.
  */
-import { Pages, Tabs, Icons } from './config/imports';
-import Privileges from './config/privileges';
+import { join } from 'path';
+import configuration from './config';
 
-export default {
-    name: '@ergo/activity-logs',
-    type: 'page',
-    moduleRelations: [
-        '@ergo/users',
-    ],
-    privileges: Privileges,
-    nuxt: {
-        aliases: {
-            '@ActivityLogs': '/',
-        },
-    },
-    router: [
-        {
-            name: 'activity-logs',
-            path: '/activity-logs',
-            component: Pages.UsersActivityLogs,
-            meta: {
-                access: true,
-                title: 'Activity logs',
-                group: {
-                    title: 'System',
-                    menuPosition: 1000,
-                    icon: Icons.Settings,
-                },
-                isMenu: true,
-                menuPosition: 3,
-                privileges: {
-                    namespace: Privileges.USER.namespace,
-                    read: Privileges.USER.read,
-                },
-                redirectTo: 'activity-logs-grid',
-            },
-            children: [
-                {
-                    name: 'activity-logs-grid',
-                    path: 'grid',
-                    component: Tabs.UsersActivityLogsGridTab,
-                    meta: {
-                        title: '',
-                        breadcrumbs: [
-                            {
-                                title: 'System',
-                                icon: Icons.Settings,
-                            },
-                        ],
-                        privileges: [],
-                    },
-                },
-            ],
-        },
-    ],
-};
+export default async function () {
+    const {
+        name,
+        type,
+        aliases = {},
+    } = configuration;
+
+    if (!this.options.ergoModules) this.options.ergoModules = {};
```

Comment:
this inside js file? What does it refer to?

Reply:
It is basic nuxt module and `index.js` is main module file. You can directly access Nuxt options using `this` reference.
Doc: https://nuxtjs.org/guide/modules/#write-a-basic-module